builder.go

```diff
@@ -100,7 +100,7 @@ func (b Builder) Build(ctx context.Context, outputFile string) error {
 	// compile
 	cmd := buildEnv.newCommand("go", "build",
 		"-o", absOutputFile,
-		"-ldflags", "-w -s", // trim debug symbols
+		"-ldflags", b.osEnvOrDefaultValue("XK6_BUILD_LD_FLAGS", "-w -s"), // trim debug symbols by default
```

Comment:
Can you change this to overwrite the whole 'ldflags' thing and in practice implement https://github.com/grafana/xk6/issues/4 ? So renaming this to `XK6_BUILD_FLAGS`

I would also argue we can drop the whole `-ldflags` by default?

Reply:
Hi @mstoykov , thanks for the reply!

I think what you said is a better solution and I've changed what you asked, but I'm not comfortable with removing the default values from -ldflags and the XK6_RACE_DETECTOR as described in #4. So I tried to leave those things with retro compatible behavior.


I was able to build with xk6 from my branch with something like these (on my Makefile):


```VERSION_DETAILS := -X go.k6.io/k6/lib/consts.VersionDetails=$$(date -u +"%FT%T%z")/$(ORG_NAME)/$(PROJECT_NAME)@$$(git describe --tags --always --long --dirty)

BUILD_COMAND := xk6 build --with github.com/$(ORG_NAME)/$(PROJECT_NAME)="."

RELEASE_ENV := XK6_BUILD_FLAGS="-ldflags=-w -s $(VERSION_DETAILS)-release"

DEBUG_ENV := XK6_BUILD_FLAGS="-ldflags=-compressdwarf $(VERSION_DETAILS)-debug"

release:
	$(RELEASE_ENV) $(BUILD_COMAND) --output k6-release-$$(go env GOOS)-$$(go env GOARCH)

debug:
	$(DEBUG_ENV) $(BUILD_COMAND) --output k6-debug-$$(go env GOOS)-$$(go env GOARCH)```

Reply:
Hey, sorry about the late response, but I'm not sure implementing #4 is just a matter of renaming this env var.

`-ldflags` is a composite flag that is passed to `go tool link` and accepts spaces. So to avoid the ambiguity when other `go build` flags are specified, it should ideally be enclosed in quotes, e.g. `-ldflags='-w -s'`. But the current way this is passed to `buildEnv.newCommand()` will fail in confusing ways.

Consider:
- `XK6_BUILD_FLAGS='-race -buildvcs=false'`
  Fails with: `flag provided but not defined: -race -buildvcs`
- `XK6_BUILD_FLAGS='-buildvcs=false -ldflags="-s -w" -race'`
  Fails with: `invalid boolean value "false -ldflags=\"-s -w\" -race" for -buildvcs: value is neither 'auto' nor a valid bool`
  (`-race` is also ambiguous since it's both a `go build` and `go tool link` flag :face_exhaling:)
- `XK6_BUILD_FLAGS='-ldflags="-s -w" -race -buildvcs=false'`
  Fails with: `invalid value "\"-s -w\" -race -buildvcs=false" for flag -ldflags: parameter may not start with quote character "`

So currently `XK6_BUILD_FLAGS` is not very useful besides for setting a single flag at a time.

I'd suggest to either:
- revert this change, and make it work for `XK6_BUILD_LD_FLAGS` only,
- or, preferably, do something more sophisticated for parsing `XK6_BUILD_FLAGS`, and pass the flags as a slice to `buildEnv.newCommand()`, so that it supports more than one flag.

WDYT?

Reply:
Nice catch, @imiric! Thank you for your input!

In my utilization above I only used one flag each time and did not notice what you said.

I've added a new change to my branch that seems to solve the problem in a preferable way (or at least I tried): I was able to build with the three cases that you mentioned with that and my own build is working fine, too...

So, what do you think?